src/main/java/org/jitsi/videobridge/Conference.java

```diff
@@ -1146,6 +1147,11 @@ public ConferenceShim getShim()
         return shim;
     }
 
+    /** The maximum number of packet handlers we want to execute in one task.
+     *   TODO: this number is pulled out of the air; tune it.
+     */
+    public static final int MAX_HANDLERS_PER_TASK = 5;
```

Comment:
I suspect this may depend on the number of cores a machine has. Shall we make it configurable?

Reply:
Possibly; this was just a quick and dirty check to see how much difference this made.

Reply:
We already populate the `CPU_POOL` based on the number of cores, so if anything maybe we'd want to base this on the size of the cpu pool, so we only make this decision in one place.

Reply:
Looking at YourKit, `AbstractExectorService.submit` seems to spend a fair amount of its time in `LinkedBlockingQueue.offer` which has to acquire a lock, so we may well want more tasks than cores here.

Reply:
Hm, we probably just shouldn't be using `offer` there.

Reply:
That's inside the Java implementation -- and never mind, I was looking at putting things on the executor queue, not the execution of the queues.

Reply:
Oh, I see what you mean now.  👍